runtime/src/bank.rs

```diff
@@ -2932,13 +2932,14 @@ impl Bank {
                 invalid_vote_keys.insert(vote_pubkey, InvalidCacheEntryReason::WrongOwner);
                 return None;
             }
-            let vote_state = match vote_account.vote_state().deref() {
-                Ok(vote_state) => vote_state.clone(),
-                Err(_) => {
+            let vote_state = vote_account
+                .vote_state()
+                .as_ref()
+                .map_err(|_| {
```

Comment:
Does using `map_err` here actually release the lock?

Reply:
I would say yes because [`map_err`](https://doc.rust-lang.org/stable/std/result/enum.Result.html#method.map_err) consumes `self`